cloud-controller-manager/do/loadbalancers.go

```diff
@@ -322,6 +346,21 @@ func getCertificateIDFromLB(lb *godo.LoadBalancer) string {
 // Load Balancer.
 func (l *loadBalancers) recordUpdatedLetsEncryptCert(ctx context.Context, service *v1.Service, lbCertID, serviceCertID string) error {
 	if lbCertID != "" && lbCertID != serviceCertID {
+		if serviceCertID != "" {
```

Comment:
Not sure if we want this here because this method is used to update the service (annotation) cert ID if it doesn't match the one on the LB, so I think this addition will short circuit that check since if it doesn't match then it's likely rotated out of the account and so won't exist? 🤔

Reply:
This particular check will only short circuit out of checking for existence of the `serviceCert` if a value does not exist and then it will fall through to see if there is a valid certificate on the lb below at line 364. 

It's probably worth relooking at how this method works in conjunction with `ensureCertificate` to see if there is any room for collapsing.